org.civicrm.reqangular/src/common/decorators/uib-tooltip.decorator.js

```diff
@@ -0,0 +1,150 @@
+/* eslint-env amd */
+
+define([
+  'common/angular',
+  'common/angularBootstrap'
+], function (angular) {
+  'use strict';
+
+  var $document, $timeout;
+
+  return ['$delegate', '$document', '$timeout', function ($delegate, _$document_, _$timeout_) {
+    $document = _$document_;
+    $timeout = _$timeout_;
+
+    return function () {
+      var tooltip = $delegate.apply(this, arguments);
+      var compilator = tooltip.compile;
+      var isTouchDevice = 'ontouchstart' in $document[0].documentElement;
+      var clickType = isTouchDevice ? 'touchend' : 'click';
+
+      tooltip.compile = function (tElement, tAttr) {
+        return function link (scope, $element, attributes, tooltipCtrl) {
+          var $overlay;
+          var isTooltipClickable = $element.attr('uib-tooltip-clickable');
+
+          isTooltipClickable && setInitialAttributesForClickableTooltip();
+          compilator(tElement, tAttr)(scope, $element, attributes, tooltipCtrl);
+          isTooltipClickable && makeTooltipClickable();
+
+          /**
+           * Injects an overlay element to the original tooltip trigger element
+           */
+          function injectOverlayToTriggerElement () {
+            $overlay = angular.element('<div class="tooltip-overlay"></div>');
+
+            $element.css({ 'position': 'relative' });
+            $overlay.css({
+              position: 'absolute',
+              height: '100%',
+              width: '100%',
+              'z-index': 1,
+              top: 0,
+              left: 0
+            });
+            $element.append($overlay);
+          }
+
+          /**
+           * Makes the tooltip clickable
+           * by making it hoverable on non-touch devices
+           * and dismissable on touch devices
+           */
+          function makeTooltipClickable (attributes) {
+            isTouchDevice && injectOverlayToTriggerElement();
+            setEventHandlersToTriggerElements();
+          }
+
+          /**
+           * Sets handlers to the tooltip trigger elements
```

Comment:
*Defines the handlers that will be used to display the tooltip depending on the device*

Reply:
Done.